pkg/upload/upload.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package upload
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/collection/entry"
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/file/splitter"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	multiPartFormData = "multipart/form-data"
+	EncryptHeader     = "swarm-encrypt"
+)
+
+// FileUploadInfo contains the data for a file to be uploaded
+type FileUploadInfo struct {
+	FileName    string
+	FileSize    int64
+	ContentType string
+	ToEncrypt   bool
+	Reader      io.Reader
+}
+
+// GetFileHTTPInfo extracts file info for upload from HTTP request
+func GetFileHTTPInfo(r *http.Request) (*FileUploadInfo, error) {
```

Comment:
extracted from the current implementation of `pkg/api/file.go/fileUploadHandler`